### ISSUE TITLE
Files with long names are packed into the wrong directory when using `pkg_tar` with `deps` and a `package_dir_file` set

### ISSUE THREAD

**Issue by alexjpwalker** [bazelbuild/rules_pkg#250]
## Description of the problem

Files with long names (~100+ chars) are packed into the wrong directory when using `pkg_tar` with a `package_dir_file` set.

## Reproducible example

- Create a Bazel project with the following package structure:
```
files
|___ lorem-ipsum-dolor-sit-amet-consectetur-adipiscing-elit-fusce-laoreet-lorem-neque-sed-pharetra-erat.txt
|___ python.txt
src
|___ bazel.bzl
|___ BUILD
|___ package-dir.txt
```
- Leave the two files in `files` empty.
- Set the content of `src/package-dir.txt` to: `package`
- In `src/bazel.bzl`, load `pkg_tar` and create a single def:
```
def make_a_targz(name, deps):
    pkg_tar(
        name = name,
        deps = deps,
        extension = "tar.gz",
        package_dir_file = "package-dir.txt",
    )
```
- Set the content of `src/BUILD` to:
```
load("//src:bazel.bzl", "make_a_targz")

make_a_targz(
    name = "bad-targz",
    deps = ["files.tar.gz"],
)
```
- In the root directory of the project, run `tar -czf src/files.tar.gz files`
- Then run `bazel build //src:bad-targz`
- After building, run `tar -tf bazel-bin/src/bad-targz.tar.gz`

## Expected output

All files should be listed in a root directory named `package`

## Actual output

The long filename was not correctly repackaged.
```
./
./package/
./package/files/
./package/files/python.txt
files/lorem-ipsum-dolor-sit-amet-consectetur-adipiscing-elit-fusce-laoreet-lorem-neque-sed-pharetra-erat.txt
```

## Environment

This issue is reproducible on Mac OS X Mojave and Linux Ubuntu 20.04, running Bazel version 3.3.1 and `pkg_tar` version 0.2.5.

**Comment by aiuto** [comment]
This looks like a duplicate of the problem with PAX format.

**Comment by alexjpwalker** [comment]
https://github.com/bazelbuild/rules_pkg/issues/216 ?

**Comment by RafaelMarinheiro** [comment]
This happens particularly often when you want to repackage another pkg_tar in an inner folder.

This is happening because the tarfile.TarInfo from the original file is being reused and its pax headers still point to the old path. Cleaning up the 'path' header before writing works because the `tarfile` library writes it again if needed.

**Comment by aiuto** [comment]
Believed fixed by #326

**Comment by vmax** [comment]
@aiuto would it be possible to cut a release with this fix included soon?

**Comment by aiuto** [comment]
Yes. I am just waiting for a few cleanup PRs to get merged.
@nacl Do you have any RPM changes that are half in place that would block a
clean release?

**Comment by nacl** [comment]
> @nacl Do you have any RPM changes that are half in place that would block a
> clean release?

Not so much with RPM stuff, but IMO, a "clean" release would be blocked on adding conceptual documentation of the `pkg_filegroup` framework.  `pkg_rpm` can now directly use it, and it should thus be given proper treatment.

I'll try to do this in the next few days, but my schedule is not stable over the next few weeks; I may not able to get to writing docs in a concerted manner until the end of the month.

In the meantime, it should be safe to clone the repository at the desired commit and load the ruleset from there.

**Comment by aiuto** [comment]
On Wed, May 12, 2021 at 12:19 PM Andrew Psaltis ***@***.***>
wrote:

> @nacl <https://github.com/nacl> Do you have any RPM changes that are half
> in place that would block a
> clean release?
>
> Not so much with RPM stuff, but IMO, a "clean" release would be blocked on
> adding conceptual documentation of the pkg_filegroup framework. pkg_rpm
> can now directly use it, and it should thus be given proper treatment.
>
I don't think we have to block on that. The legacy RPM still works fine. As
long as that is the default, you can continue to work on the new one.

> I'll try to do this in the next few days, but my schedule is not stable
> over the next few weeks; I may not able to get to writing docs in a
> concerted manner until the end of the month.
>
Don't rush. I would rather do a release now with current behavior and then
we can spend the next month refining pkg_filegroup.  I'm working on
pkg_files for pkg_tar, so I have a fairly large example use case in the
works. As I have been working, I think introducing pkg_files to that will
result in potential deprecations and/or breaking changes to pkg_tar, so
this may be a good poit to have a frozen release.

**Comment by nacl** [comment]
Fair enough.  I'll work on the outstanding cleanup reviews today/tomorrow.
